### PR TITLE
Adding Jetstream support to locust solution.

### PR DESCRIPTION
**Pull request by kfswain**
In addition to Jetstream support:

- Made --stop-timeout an opt-in variable
- Piped namespace var to locust master KNS name
- Bumped transformers version for Gemma support

### CODE REVIEW

File: benchmarks/benchmark/tools/locust-load-inference/variables.tf [hunk 1]
```diff
@@ -71,6 +71,12 @@ variable "num_locust_workers" {
   default     = 1
 }
 
+variable "stop_timeout" {
+  description = "Length of time before a locust job is stopped."
+  type        = number
+  default     = 0
```
Comment (annapendleton) [inline benchmarks/benchmark/tools/locust-load-inference/variables.tf:77]: Default should be 10800 to keep consistent behavior. Or is 0 a more reasonable default in this case?

File: benchmarks/benchmark/tools/locust-load-inference/locust-docker/locust-tasks/run.sh [hunk 2]
```diff
@@ -24,17 +24,20 @@ if [[ "$LOCUST_MODE" = "master" ]]; then
     # For inferencing workloads with large payload having no wait time is unreasonable.
     # This timeout is set to large amount to avoid user tasks being killed too early.
     # TODO: turn timeout into a variable.
-    LOCUS_OPTS="$LOCUS_OPTS --master --stop-timeout 10800"
+    LOCUST_OPTS="$LOCUST_OPTS --master "
+    if [[ "$STOP_TIMEOUT" != 0 ]]; then
+        LOCUST_OPTS="$LOCUST_OPTS --stop-timeout $STOP_TIMEOUT"
```
Comment (annapendleton) [inline benchmarks/benchmark/tools/locust-load-inference/locust-docker/locust-tasks/run.sh:29]: OOC, why does this need to be an opt-in var? Is this related to making the runner work?

File: benchmarks/benchmark/tools/locust-load-inference/locust-docker/locust-tasks/tasks.py [hunk 3]
```diff
@@ -123,6 +128,8 @@ def get_token_count(prompt, resp):
             tokenizer.encode(resp_dict['text_output']))
     elif backend == "sax":
         number_of_output_tokens = 0  # to be added
+    elif backend == "jetstream":
+        number_of_output_tokens = 0
```
[the remark this answers is not in the document]
Comment (kfswain) [reply benchmarks/benchmark/tools/locust-load-inference/locust-docker/locust-tasks/tasks.py:132]: Yeah, I was just adding barebones support and fixes. Will follow up with a metrics-related PR

File: benchmarks/benchmark/tools/locust-load-inference/variables.tf (hunk 1 above)
Comment (kfswain) [reply benchmarks/benchmark/tools/locust-load-inference/variables.tf:77]: I chose `0`, as that forces this flag to be opt-in. My thinking is making the field opt-in reduces the chance of hitting: https://github.com/locustio/locust/issues/2621. Which I hit during development

LMKWYT

File: benchmarks/benchmark/tools/locust-load-inference/locust-docker/locust-tasks/run.sh (hunk 2 above)
Comment (kfswain) [reply benchmarks/benchmark/tools/locust-load-inference/locust-docker/locust-tasks/run.sh:29]: Addressed in the previous comment

File: benchmarks/benchmark/tools/locust-load-inference/locust-docker/locust-tasks/tasks.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (annapendleton) [reply benchmarks/benchmark/tools/locust-load-inference/locust-docker/locust-tasks/tasks.py:132]: Sounds good ty for clarifying!

File: benchmarks/benchmark/tools/locust-load-inference/variables.tf (hunk 1 above)
Comment (annapendleton) [reply benchmarks/benchmark/tools/locust-load-inference/variables.tf:77]: I tested locally with 0 with my current tests and confirmed it has no negative impact. Also looks like stop_timeout doesn't do anything in terms of stopping long running user tasks as I had thought - it actually impacts "graceful termination" - https://github.com/locustio/locust/blob/a2f328fdb2fcf5592b07b737680a9edcfa201301/locust/env.py#L83

Comment (annapendleton) [reply benchmarks/benchmark/tools/locust-load-inference/variables.tf:77]: With that - I'm fine with this default set to 0 :) thanks for explaining and fixing this!

**Review by annapendleton — APPROVED** [review]
